test/show.jl

```diff
@@ -1462,3 +1462,10 @@ end
 # issue #30927
 Z = Array{Float64}(undef,0,0)
 @test eval(Meta.parse(repr(Z))) == Z
+
+# issue #31065, do not print parentheses for nested dot expressions
+let
+    struct I31065 x::Union{I31065,Nothing} end
+    foo = I31065(I31065(nothing))
```

Comment:
The struct definition and instance are not necessary to print the expression --- the expression is not evaluated.

Reply:
Yes of course. Thanks for rescuing me, again!